keep-ui/app/settings/add-user-modal.tsx

```diff
@@ -0,0 +1,181 @@
+import { Dialog } from '@headlessui/react';
+import { useState } from 'react';
+import Select, { components, SingleValue, MenuListProps} from 'react-select';
+import { TextInput, Button, Subtitle } from '@tremor/react';
+import { AuthenticationType } from "utils/authenticationType";
+import { User } from "./models";
+import { getApiURL } from "utils/apiUrl";
+
+interface RoleOption {
+    value: string;
+    label: string | JSX.Element;
+    tooltip?: string;
+    isDisabled?: boolean;
+  }
+
+interface AddUserModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  authType: string;
+  setUsers: React.Dispatch<React.SetStateAction<User[]>>;
+  accessToken: string;
+}
+
+export default function AddUserModal({ isOpen, onClose, authType, setUsers, accessToken }: AddUserModalProps) {
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [selectedRole, setSelectedRole] = useState<SingleValue<RoleOption>>(null);
+  const [password, setPassword] = useState('');
+  const [hoveredOption, setHoveredOption] = useState<string | null>(null);
+  const [addUserError, setAddUserError] = useState('');
+
+  let roleOptions: RoleOption[] = [
+    { value: 'admin', label: 'Admin', tooltip: "Admin has read/write/update/delete for every resource" },
+    { value: 'noc', label: 'NOC', tooltip: "NOC has the ability to view alerts and assign to alerts" },
+    { value: 'create_new', label: 'Create custom role', isDisabled: true, tooltip: "For custom roles, contact Keep team" },
+];
+
+const CustomOption = (props: any) => {
```

Comment:
what is this exactly?

Reply:
this is to customize the options so that they'll have tooltip

Reply:
https://stackoverflow.com/questions/59154960/how-to-add-icons-and-tooltip-on-react-select-component